tls/s2n_psk.h

```diff
@@ -46,6 +46,7 @@ struct s2n_psk {
 };
 S2N_RESULT s2n_psk_init(struct s2n_psk *psk, s2n_psk_type type);
 S2N_CLEANUP_RESULT s2n_psk_wipe(struct s2n_psk *psk);
+S2N_RESULT s2n_psk_clone(struct s2n_psk *new_psk, struct s2n_psk *original_psk);
```

Comment:
Isn't the convention we use in s2n the other way round - original psk first and clone psk second?   
`S2N_RESULT s2n_psk_clone(struct s2n_psk *original_psk, struct s2n_psk *new_psk);`

Ex: https://github.com/aws/s2n-tls/blob/main/utils/s2n_mem.h#L31

Reply:
Is there an S2N convention for this? It looks like stuffers and blobs do (source, destination), but the constant time copy does (destination, source). I followed [memcpy](http://www.cplusplus.com/reference/cstring/memcpy/) pattern of (destination, source).

Does it matter?

Reply:
I think we should stay consistent and follow what the stuffers and blobs do as they are a part of the core data structures within s2n. The constant time copies seem like an anti-pattern to me and I wouldn't want to add to it.

Reply:
The constant time copy matches memcpy though. I'd argue matching memcpy is going to make it more intuitive for developers, and the stuffer and blob copies (which are pretty rarely used) are the anti-pattern.

Reply:
I believe we should set an S2N convention for function argument evaluation order ( either R->L or L->R),  as C doesn't have specify a defined behavior for it. 

The reason for it would be ease of use and readability of APIs. If we use L->R in some places and R->L in other locations, the developer is left to analyze the name of the argument (not all our function argument names are intuitive) or  as in most cases refer to the logic to identify which arguments are input and which are outputs for that function call, which I believe will not be a good experience.